app/models/flood_risk_engine/enrollment.rb

```diff
@@ -68,6 +72,8 @@ def state_machine
     def submit
       return if submitted_at.present?
       self.submitted_at = Time.zone.now
+      self.reference_number = ReferenceNumber.create
+      save
```

Comment:
Does it need a transaction around it? Ref number might get created ok no line 75 but save on line 76 fails for some reason. Unlikely I know but would prevent gaps in the sequence